Assets/GameAssets/Player/PlayerScripts/PlayerInteraction.cs

```diff
@@ -0,0 +1,108 @@
+/**********************************************************************************************************************
+// File Name :         PlayerInteraction.cs
+// Author :            Alex Kalscheur
+// Creation Date :     11/10/24
+// 
+// Brief Description : Controls the interaction system on the player side
+**********************************************************************************************************************/
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+[RequireComponent(typeof(PlayerInput))]
+
+/// <summary>
+/// Controls the interaction system on the player side
+/// </summary>
+public class PlayerInteraction : MonoBehaviour
+{
+    [Tooltip("How far can the player reach interactable objects")]
+    [SerializeField] private float _maxReach;
+    [Tooltip("Ray that is cast from camera to find interactable objects")]
+    Ray ray;
+
+    //Input
+    private PlayerInput _playerInput;
+    private InputAction _interactInput;
+
+
+    /// <summary>
+    /// Every frame, Ray is cast.  If the ray hits something and the button is
+    /// </summary>
+    private void Update()
+    {
+        ray = Camera.main.ViewportPointToRay(new Vector3(0.5f, 0.5f, 0));
+        if (Physics.Raycast(ray, out RaycastHit hit, _maxReach))
+        {
+            GiveInteractableFeedback(hit.collider.gameObject);
+            if (_interactInput.WasPerformedThisFrame()) 
```

Comment:
Perhaps this doesn't need to be done in update. I would make the input call a function when pressed and then it checks the status of ray and if that hit something then it calls interact

Reply:
We need some way for there to be a check if the player is looking at something interactable for UI purposes

Reply:
Yeah I agree the raycast needs to happen every frame, but does the WasPerformedThisFrame need to?